src/clib/pioc_support.c

```diff
@@ -2368,7 +2368,7 @@ int PIOc_openfile_retry(int iosysid, int *ncidp, int *iotype, const char *filena
             return check_mpi(file, mpierr, __FILE__, __LINE__);
         if ((mpierr = MPI_Bcast(pio_type_size, nvars, MPI_INT, ios->ioroot, ios->my_comm)))
             return check_mpi(file, mpierr, __FILE__, __LINE__);
-        if ((mpierr = MPI_Bcast(mpi_type, nvars, MPI_INT, ios->ioroot, ios->my_comm)))
+        if ((mpierr = MPI_Bcast(mpi_type, nvars*(int)(sizeof(MPI_Datatype)/sizeof(int)), MPI_INT, ios->ioroot, ios->my_comm)))
```

Comment:
I believe this should be:
MPI_Bcast(mpi_type, nvars, MPI_Datatype, ios->ioroot, ios->my_comm);

Reply:
The third argument to `MPI_Bcast` should be the size of the elements in the buffer that is the first argument. As far as I know, there isn't an MPI datatype definition for `MPI_Datatype` (that is, `MPI_INT` is use for `int`, `MPI_FLOAT` is used for `float`, etc., but I don't know of anything that is used for `MPI_Datatype`). My solution was to treat the size of the buffer as `MPI_INT`, but to set the size of the buffer as the normal size scaled by the ration of `sizeof(MPI_Datatype)/sizeof(int)`.

Reply:
OK, I get what you are saying but this is a rather convoluted and non-obvious way to handle it.

I think a better way would be to cast the value to long int and use the MPI_LONG_INT type.

Reply:
That would require an if block - maybe we just need some documentation at this line explaining the need for convolution?